samcli/lib/package/packageable_resources.py

```diff
@@ -121,6 +121,7 @@ def export(self, resource_id: str, resource_dict: Optional[Dict], parent_dir: st
         temp_dir = None
         if is_local_file(property_value) and not is_zip_file(property_value) and self.FORCE_ZIP:
             temp_dir = copy_to_temp_dir(property_value)
+            assert self.PROPERTY_NAME is not None
```

Comment:
Potential `AssertionError`, but actually this function fails with `self.PROPERTY_NAME is None` earlier.

This assert could be moved higher, because `jmespath.search(None)` fails as well. Probably a proper fix would be to change `PROPERTY_NAME: Optional[str] = None` to `PROPERTY_NAME: str = ""`

Reply:
I made a change that I described above. `assert` was removed, `PROPERTY_NAME` is now always an `str`.